docs/guides/hooks-list.rst

```diff
@@ -543,9 +543,11 @@ Routing
 **route:config, <route_name>**
 	Allows altering the route configuration before it is registered.
 	This hook can be used to alter the path, default values, requirements, as well as to set/remove middleware.
+	Please note that the handler for this hook should be registered outside of the ``init`` event handler, as core routes are registered during ``plugins_boot:before`` event.
```

Comment:
I see the registration for `plugins_boot`, not `plugins_boot:before`

Reply:
triggerSequence calls before and after events

Reply:
correct, but since you register the hook on `plugins_boot` it won't be registered until then. https://github.com/hypeJunction/Elgg-leaf/blob/6afebe0acf5bed28621237f973062a867d4bf474/engine/classes/Elgg/Application.php#L227-L228

Reply:
That's the intention. 
start.php is loaded during `plugins_boot:before`, plugin register a handler for `route:config`, which is then executed when `plugins_boot` runs. I put it high up to keep the priority high.

Reply:
The way I envisioned this when I rewrote the boot process, is allow plugins to subscribe for `plugins_boot` event from `start.php` in order to get things done before `init`, e.g. to configure some core service, without putting runtime logic into the `start.php` itself.

Before, you would have to do something like this:

```php
elgg_set_email_transport(); // really bad for testing
elgg_register_event_handler('init', 'system');
```

Now:

```php
elgg_register_event_handler('plugins_boot', 'system', function() {
   elgg_set_email_transport();
});
elgg_register_event_handler('init', 'system');
```

Reply:
As a next step, I would like to get rid of `start.php` entirely.

```php
// elgg-plugin.php
return [
   'service' => MyPlugin::class,
];
```

```php
class MyPlugin implements PluginService {

}
```

```php
interface PluginService {
   public function load();
   public function boot();
   public function init();
   public function ready();
}
```

Reply:
I read the documentation as follows
Core registers it's routes in `plugins_boot:before` `system` hook. So if I as a dev want te be able to change those routes i need to register my custom handler before this. Which hook comes before this, nothing.

In reality core registers it's routes in `plugins_boot` ``system`, so i can register my registration of the route handler in `plugins_boot:before` `system`

Reply:
Oh, I see now. I initially put them in the before event, that didn't work, and I didn't update the docs. Sorry, got your point now.

Reply:
It would be so nice if there were no `start.php` with all the junk. We could do things right...